clients/web/src/components/tauri/configuration-value.tsx

```diff
@@ -1,57 +1,43 @@
-import { Show, For, Switch, Match } from "solid-js";
+import { Show, For, createMemo } from "solid-js";
 import { ConfigurationTooltip } from "./configuration-tooltip";
 import { Flags } from "./configuration-value/flags";
+import { TauriConfig } from "~/lib/tauri/config/tauri-conf";
 
-type ConfigurationValue =
-  | ConfigurationRecord
-  | string
-  | []
-  | boolean
-  | null
-  | undefined
-  | unknown;
-
-/* eslint-disable-next-line @typescript-eslint/no-empty-interface */
-interface ConfigurationRecord extends Record<string, ConfigurationValue> {}
+type ConfigurationValue = TauriConfig[keyof TauriConfig];
 
 interface ConfigurationValueProps {
   parentKey: string;
   key: string;
   value: ConfigurationValue;
 }
 
-interface TextConfigurationValueProps extends ConfigurationValueProps {
+type TextConfigurationValueProps = Omit<ConfigurationValueProps, "value"> & {
   value: string | boolean;
-}
+};
 
 interface ArrayConfigurationValueProps extends ConfigurationValueProps {
-  value: [];
+  value: ConfigurationValue[];
 }
 
 interface ObjectConfigurationValueProps extends ConfigurationValueProps {
-  value: ConfigurationRecord;
+  value: Omit<ConfigurationValue, "undefined">;
 }
 
 export function ConfigurationValue(props: ConfigurationValueProps) {
-  return (
-    <Switch
-      fallback={<TextValue {...(props as TextConfigurationValueProps)} />}
-    >
-      <Match
-        when={
-          typeof props.value === "string" || typeof props.value === "boolean"
-        }
-      >
-        <TextValue {...(props as TextConfigurationValueProps)} />
-      </Match>
-      <Match when={Array.isArray(props.value)}>
-        <ArrayValue {...(props as ArrayConfigurationValueProps)} />
-      </Match>
-      <Match when={typeof props.value === "object" && props.value !== null}>
-        <ObjectValue {...(props as ObjectConfigurationValueProps)} />
-      </Match>
-    </Switch>
-  );
+  const selectedTemplate = createMemo(() => {
```

Comment:
I see what you are trying to do here, solidjs and typeguard are a bit of a pain… Careful here, you might break the reactivity.

You can craft custom helper for this like this:

```ts
function isTextValue(value: any): string | false {
if (typeof value === "string") return value;

return false;
}

// Usage
<Match when={isTextValue(props.value)}>
  {v => <TextValue value={v()} /> // v() is a string here}
</Match>
```

We can pair on it if you want ;)

Reply:
I agree that your solution is also workable. For some reason I tried something similar before and ran into problems. 

The solution I setup should not break reactivity? It is just a computed function that will get reevalutated if `props` or `props.value` changes. Since the function gets called in the JSX it should be fine. (I think in general our setup would also produce a warning if I butchered the reactive scope by using reactive variables in a non reactive scope)